### PR TITLE
Simplify Checkstyle configuration

### PR DESCRIPTION
**Pull request by artembilan**
* Move `checkstyle-conventions.gradle` content to the `build.gradle` since it is short enough
* Remove `checkstyle-header.txt` and its handling in favor of Spring Javaformat out-of-the box logic
* Upgrade to Gradle `8.5`
* Remove suppression for `Javadoc*` checks since it is handled by Spring Javaformat
* Leave only `MissingJavadocMethodCheck` exclusion from `SpringChecks` since all others really make sense for our code quality

### CODE REVIEW

**Comment by artembilan** [comment]
It looks like Spring Javaformat has to be upgraded to the latest Checkstyle to make them all compatible with whatever we expect.
I mean that even Gradle `8.5` doesn't help 🤷

**Comment by artembilan** [comment]
It looks like `burrunan/gradle-cache-action` scans console output for `error` and fails the build.
Unlike whatever we have with the `jf gradle` for SNAPSHOT build and when we run locally.
The `ignoreFailures = true` indeed has an effect, but with `burrunan/gradle-cache-action` for PRs it fails gathering all the errors in the end for final failure 🤷 : 
```
BUILD SUCCESSFUL in 7m 28s
515 actionable tasks: 394 executed, 121 from cache
cacheid: 1737
cacheid: 1738
Cache Size: ~0 MB (5774 B)
Cache Size: ~0 MB (4638 B)
Error: [Task :spring-aggregator-function:checkstyleMain] [JavadocPackage] Missing package-info.java file.
Error: [Task :spring-aggregator-function:checkstyleMain] [SpringMethodVisibility] Method 'messageStore' in private class should not be public.
```

File: build.gradle [hunk 1]
```diff
@@ -138,6 +139,22 @@ configure(javaProjects) { subproject ->
 
 	[compileJava, compileTestJava]*.options*.compilerArgs = ['-Xlint:all,-options,-processing', '-parameters']
 
+	checkstyle {
+		toolVersion = '10.3'
```
Comment (onobc) [inline build.gradle:143]: Now that we are on Gradle 8.5 can we bump this to the latest and see if we can ditch the dependency on 150?

Comment (artembilan) [reply build.gradle:143]: Yeah... Sorry for not being very clear with my previous comment about Spring Javaformat and Gradle `8.5`.

I tried all those possible combinations: as you pointed out only this one works 🤷 

Feel free to play with that on your own: perhaps I still have missed something.

Thanks

**Review by onobc — APPROVED** [review]
LGTM thanks @artembilan

**Pull request merged**